### PR TITLE
Ci/CD Integration guide

### PR DESCRIPTION
**Pull request by aj-jester**
First pass ready for review.

Related `kotsapps` PR https://github.com/replicatedhq/kotsapps/pull/27

### CODE REVIEW

File: content/vendor/guides/internal-starter-repo.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+cp -R kotsapps/starter-repo/k8s/* kots-app/
+```
+
+Commit and push your `kots-app` repo from the `master` branch.
+
+```shell
+$ cd kots-app
+$ git add .
+$ git commit -m "New Kots App"
+$ git push
```
Comment (dexhorthy) [inline content/vendor/guides/internal-starter-repo.md:60]: Let's not use the `$` prefix for multiline command blocks, per the style guide: https://github.com/replicatedhq/help-center/blob/master/CONTRIBUTING.md#multi-line-code-snippets-should-not-have-the-bash-prompt

File: content/vendor/guides/ci-cd-integration.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+It is broken into three sections:
+
+- [Environment Variables](#environment-variables)
+- [Kubernetes Objects](#kubernetes-objects)
+- [Helm Chart](#helm-chart)
+
+## Environment Variables
+
+This section will assume you've already completed the [Automating Your Workflow](/vendor/guides/quickstart/#automating-your-workflow) and set up a local git repo to manage your application.
```
Comment (dexhorthy) [inline content/vendor/guides/ci-cd-integration.md:18]: This line doesn't really seem to mesh with the rest of the article. Is the goal of this article to get me to a local git repo?

File: content/vendor/guides/ci-cd-integration.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+Follow the steps below if you are just working with plain-YAML kubernetes API objects.
+
+- In the root directory of the internal repository add this `Makefile`
+
+```shell
+curl -LS https://raw.githubusercontent.com/replicatedhq/replicated-starter-kots/master/Makefile -o ./Makefile
+```
+
+- Make a copy of [manifests](https://github.com/replicatedhq/replicated-starter-kots/tree/master/manifests) directory from the KOTS starter repo and place it in your internal repo's root directory.
```
Comment (dexhorthy) [inline content/vendor/guides/ci-cd-integration.md:35]: what if I already have my manifests in another folder like `k8s` or `deploy` -- does this mean I have to re-organize my repo in order to make it work with kots?

File: content/vendor/guides/ci-cd-integration.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+- Execute the following `make` command to deploy
+```shell
+make release
+```
+
+- Verify release in [vendor.replicated.com](https://vendor.replicated.com).
+
+## Helm Chart
+
+The steps below show how to package Helm Templates into a `.tgz` Chart and deploy as a Kots App. This section assumes you've completed the [Packaging a Helm Chart](/vendor/guides/helm-chart) guide.
```
Comment (dexhorthy) [inline content/vendor/guides/ci-cd-integration.md:65]:
```suggestion
The steps below show how to modify a repo containing a Helm chart to be deployable as a Kots App. This section assumes you've completed the [Packaging a Helm Chart](/vendor/guides/helm-chart) guide and are familiar with how KOTS processes helm charts as part of an application.
```

File: content/vendor/guides/ci-cd-integration.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+│       └── support-bundle.yaml
+├── templates
+│   ├── config-map.yaml
+│   ├── deployment.yaml
+│   ├── ingress.yaml
+│   └── service.yaml
+└── values.yaml
+```
+
+Here is a real world [Helm Chart](https://github.com/replicatedhq/kotsapps/tree/master/helm-influxdb) example.
```
Comment (dexhorthy) [inline content/vendor/guides/ci-cd-integration.md:127]:
```suggestion
You can also check out a real-world [Helm Chart](https://github.com/replicatedhq/kotsapps/tree/master/helm-influxdb) example.
```

File: content/vendor/guides/ci-cd-integration.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+- Setup your CI/CD to run the following `make` command to deploy
```
Comment (dexhorthy) [inline content/vendor/guides/ci-cd-integration.md:129]:
```suggestion
- Configure your CI/CD to run the following `make` command on push
```

File: content/vendor/guides/ci-cd-integration.md (hunk 3 above)
Comment (dexhorthy) [reply content/vendor/guides/ci-cd-integration.md:35]: I like that the helm chart section of this guide started with an example repo structure *before* adding the kots bits -- what if we did the same here? Maybe we could assume the manifests start out in a non-standard place (i.e. not `manifests`)

Comment (aj-jester) [reply content/vendor/guides/ci-cd-integration.md:40]: Good idea, updated.

File: content/vendor/guides/ci-cd-integration.md (hunk 2 above)
Comment (aj-jester) [reply content/vendor/guides/ci-cd-integration.md]: Updated to better reflect what is actually needed, which is the API Token.

**Comment by aj-jester** [comment]
@dexhorthy This is ready for another review.

**Comment by aj-jester** [comment]
@dexhorthy This is ready for another round of review.

File: content/vendor/guides/ci-cd-integration.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+└── k8s
+    ├── config-map.yaml
+    ├── deployment.yaml
+    ├── ingress.yaml
+    └── service.yaml
+```
+
+- Make a copy of [manifests](https://github.com/replicatedhq/replicated-automation/tree/master/vendor/nginx-app/manifests) directory from the KOTS starter repo and place it in your internal repo's root directory.
+
+- Replace the files in `manifests/app` with the files in `k8s`. Sub directories are supported as well, so `k8s` directory can be put anywhere inside `manifests`.
```
Comment (dexhorthy) [inline content/vendor/guides/ci-cd-integration.md:42]: I still think this isn't quite right. Do you think there's a way for me to augment my existing k8s repo without having to move my manifests around?

Comment (aj-jester) [reply content/vendor/guides/ci-cd-integration.md:42]: Sure let me think about this. The thing here is with the Helm chart there is only 1 non-kots.io API file, the `tar.gz`. But with the normal kotsapp there are more than one non-kots app files so its a bit tricky. Lets discuss this further during the FDE standup on Wed.

**Pull request merged**